Add Power Status tests, Fix broken tags test

patthiel opened the pull request:
* Does the thing in the title, also adds a test to linodes to create a image-less Linode and assert that it does not boot.
* Updates the tags assertion due to the increase in max length for tags

Dorthu commented on test/linodes/linodes.bats:
```diff
@@ -65,13 +66,33 @@ load '../common'
+
+    until [ $(linode-cli linodes view $linode_id --format="status" --text --no-headers) = "offline" ]; do
+        echo 'still setting up'
+    done
```
This loop should probably have a timeout.  Also, adding a sleep per iteration would be friendly 😄

Dorthu commented on test/linodes/linodes.bats:
```diff
@@ -65,13 +66,33 @@ load '../common'
+        --root_pass $random_pass
+    local linode_id=$(linode-cli linodes list --format="id" --text --no-headers)
+
+    until [ $(linode-cli linodes view $linode_id --format="status" --text --no-headers) = "offline" ]; do
```
This test would mean more if you check that the status wasn't still "provisioning"

Dorthu commented on test/linodes/power-status.bats:
```diff
@@ -0,0 +1,102 @@
+    run createLinode
+    local linode_id=$(linode-cli linodes list --format="id" --text --no-headers)
+
+    until [ $(linode-cli linodes view $linode_id --format="status" --text --no-headers) = "running" ]; do
```
Again, I'd prefer a `!= "provisioning"` here

Dorthu commented on test/linodes/power-status.bats:
```diff
@@ -0,0 +1,102 @@
+
+    until [ $(linode-cli linodes view $linode_id --format="status" --text --no-headers) = "running" ]; do
+        echo 'still provisioning'
+    done
```
And a sleep in the loop would be a cool addition too

patthiel commented:
fair feedback. I added timeouts of 180 seconds for the until loops and added sleeps of 5 seconds between each request to be nicer, let me know if i should be more generous with that time between requests. 

As far as the method in the tests. I'm not really a fan of the "wait for this thing to _not_ be in state T,U,V then make an assertion that it's in state Z". That pattern could result in more flakey tests, because there's no guarantee that they'll be in state Z after the initial "NOT in T,U,V" conditionals. They could, eventually be in state Z and allowing for some temporary intermediate state that we don't care about that eventually transitions to the end state is ok, so long as we have a *timeout*, which I've now added (thx to you). What if more steps between T,U,V are added later. They'll break these tests, even though all i care about is that it gets to step Z

Dorthu commented on test/linodes/linodes.bats:
```diff
@@ -65,13 +66,33 @@ load '../common'
+
+    until [ $(linode-cli linodes view $linode_id --format="status" --text --no-headers) = "offline" ]; do
+        echo 'still setting up'
+    done
```
This loop still has no timeout or rate limit

patthiel replied:
good catch, added!

patthiel commented:
Think i addressed all the feedback here. Thanks for reviewing!
